ejb30/sec TCK failures

Jakarta EE TCK 9.1 Run:
https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1297/testReport/

Junit Test Report
[ejb30_sec-junit-report.txt](https://github.com/eclipse-ee4j/glassfish/files/6141469/ejb30_sec-junit-report.txt)

Test report shows as `java.lang.SecurityException`
```
org.omg.CORBA.UNKNOWN: WARNING: 00010002: Unknown user exception thrown by the server - exception: java.lang.SecurityException; message: Error while creating a JAAS subject credential.  vmcid: OMG  minor code: 2 completed: Maybe
	at com.sun.proxy.$Proxy185.runtimeexception(Unknown Source)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.convertThrowableToSystemException(MessageMediatorImpl.java:1743)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleThrowableDuringServerDispatch(MessageMediatorImpl.java:1694)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleThrowableDuringServerDispatch(MessageMediatorImpl.java:1658)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:229)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
	at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
	at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
	at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)
Caused by: java.lang.SecurityException: Error while creating a JAAS subject credential.
	at com.sun.enterprise.iiop.security.SecServerRequestInterceptor.receive_request(SecServerRequestInterceptor.java:478)
	at com.sun.corba.ee.impl.interceptors.InterceptorInvoker.invokeServerInterceptorIntermediatePoint(InterceptorInvoker.java:592)
	at com.sun.corba.ee.impl.interceptors.PIHandlerImpl.invokeServerPIIntermediatePoint(PIHandlerImpl.java:594)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.getServantWithPI(ServerRequestDispatcherImpl.java:304)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:169)
	... 9 more
```

server.log shows as `GSSException: Improperly formatted ASN.1 DER encoding for Oid`
```
[2021-03-15T08:40:25.201+0000] [glassfish 6.1] [SEVERE] [] [javax.enterprise.resource.corba.com.sun.enterprise.iiop.security] [tid: _ThreadID=61 _ThreadName=p: thread-pool-1; w: 1] [timeMillis: 1615797625201] [levelValue: 1000] [[
  iiop.IOexception
GSSException: Improperly formatted ASN.1 DER encoding for Oid
	at java.security.jgss/org.ietf.jgss.Oid.<init>(Oid.java:115)
	at com.sun.enterprise.iiop.security.GSSUtils.getOID(GSSUtils.java:279)
	at com.sun.enterprise.iiop.security.GSSUtils.verifyTokenHeader(GSSUtils.java:383)
	at com.sun.enterprise.iiop.security.GSSUtils.getMechToken(GSSUtils.java:334)
	at com.sun.enterprise.iiop.security.GSSUPToken.<init>(GSSUPToken.java:200)
	at com.sun.enterprise.iiop.security.GSSUPToken.getServerSideInstance(GSSUPToken.java:99)
	at com.sun.enterprise.iiop.security.SecServerRequestInterceptor.createAuthCred(SecServerRequestInterceptor.java:340)
	at com.sun.enterprise.iiop.security.SecServerRequestInterceptor.receive_request(SecServerRequestInterceptor.java:473)
	at com.sun.corba.ee.impl.interceptors.InterceptorInvoker.invokeServerInterceptorIntermediatePoint(InterceptorInvoker.java:592)
	at com.sun.corba.ee.impl.interceptors.PIHandlerImpl.invokeServerPIIntermediatePoint(PIHandlerImpl.java:594)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.getServantWithPI(ServerRequestDispatcherImpl.java:304)
	at com.sun.corba.ee.impl.protocol.ServerRequestDispatcherImpl.dispatch(ServerRequestDispatcherImpl.java:169)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequestRequest(MessageMediatorImpl.java:1529)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:1405)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleInput(MessageMediatorImpl.java:910)
	at com.sun.corba.ee.impl.protocol.giopmsgheaders.RequestMessage_1_2.callback(RequestMessage_1_2.java:192)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.handleRequest(MessageMediatorImpl.java:674)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.dispatch(MessageMediatorImpl.java:476)
	at com.sun.corba.ee.impl.protocol.MessageMediatorImpl.doWork(MessageMediatorImpl.java:2202)
	at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.performWork(ThreadPoolImpl.java:476)
	at com.sun.corba.ee.impl.threadpool.ThreadPoolImpl$WorkerThread.run(ThreadPoolImpl.java:519)
]]
```

--- a/appserver/security/ejb.security/src/main/java/com/sun/enterprise/iiop/security/GSSUtils.java
+++ b/appserver/security/ejb.security/src/main/java/com/sun/enterprise/iiop/security/GSSUtils.java
@@ -376,9 +376,11 @@ public class GSSUtils {
         if (token[index] != 0x06)
             throw new GSSException(GSSException.DEFECTIVE_TOKEN);
 
-        byte[] buf = new byte[token.length - index];
+        // add first two bytes to the MECH_OID_LEN
+        int oidlen = token[index+1] + 2;
+        byte[] buf = new byte[oidlen];
 
-        System.arraycopy(token, index, buf, 0, token.length - index);
+        System.arraycopy(token, index, buf, 0, oidlen);
 
         Oid mechoid = getOID(buf);
 

